script.js

```diff
@@ -1,39 +1,97 @@
-let currencyRate = {
-  timestamp: 1519296206, 
-  base: "EUR",
-  date: "2023-12-12", 
-  rates: {},
-};
+let currencyRate = [
+  {
+    timestamp: new Date().toLocaleDateString(),
+    base: "EUR",
+    date: "2023-12-12",
+    rates: {},
+  },
+  {
+    timestamp: new Date().toLocaleDateString(),
+    base: "USD",
+    date: "2023-12-12",
+    rates: [],
+  },
+  {
+    timestamp: new Date().toLocaleDateString(),
+    base: "DKK",
+    date: "2023-12-12",
+    rates: {},
+  },
+];
 
+const specials = {
+  EUR: {
+    USD: 2,
+    DKK: 3,
+  },
+  USD: {
+    EUR: 2,
+    DKK: 3,
+  },
+	DKK: {
+		EUR: 2,
+		USD: 3,
+	},
+}
+  
 
-const rates = currencyRate.rates;
-console.log(rates);
 
+// function displayTable(currBase, rates) {
+  function displayTable(currencyRate){
+  const rateTable = document.getElementById("rateTable");
+  rateTable.innerHTML = '';
+  const tableHeader ='<tr class="header"> <th style="width:35%;">From</th> <th style="width:35%;">To</th><th style="width:30%;">Rate</th> </tr>';
+  rateTable.insertAdjacentHTML('afterbegin', tableHeader);
+    currencyRate.map((item) => {
+      const currBaseitem = item.base;
+      const rates = item.rates;
+
+      for (const item in rates) {
+         const tableRow = document.createElement("tr");
+         const tableCol1 = document.createElement("td");
+         const tableCol2 = document.createElement("td");
+         const tableCol3 = document.createElement("td");
+         tableCol1.textContent = currBaseitem;
+         tableCol2.textContent = item;
+         tableCol3.textContent = rates[item];
+   
+         tableRow.appendChild(tableCol1);
+         tableRow.appendChild(tableCol2);
+         tableRow.appendChild(tableCol3);
+         rateTable.appendChild(tableRow);
+      }
+   })
+ }
 
 function addRate(newCurrency) {
   const listofCurr = document.getElementById("curr_output");
-   const listItem = document.createElement("option");
+  const listItem = document.createElement("option");
   listItem.value = newCurrency;
   listItem.innerHTML = newCurrency;
   listofCurr.appendChild(listItem);
 }
```

Comment:
what about pushing this logic to the next level? So we can switch between base currencies?

1. add remaining options and call `onSelect` function
```
<select class="select_curr" id="curr_input" name="curr_input" onchange="handleCurrencyChange()">
					<option value="EUR">EUR</option>
					<option value="USD">USD</option>
					<option value="DKK">DKK</option>
			</select>
```
2. change the rate options on selecting the base currency
```
function handleCurrencyChange() {
  const currencyIn = document.getElementById('curr_input').value
  const listofCurr = document.getElementById('curr_output')
  const listItem = document.createElement('option')
  listofCurr.innerHTML = ''
  const currentCurrency = currencyRate.find(
    (currObj) => currObj.base === currencyIn,
  )

  for (const item in currentCurrency.rates) {
    listItem.value = item
    listItem.innerHTML = item
    listofCurr.appendChild(listItem)
  }
}
```

at least what I would do, this is not mandatory :)